src/renderers/dom/server/ReactServerRenderingTransaction.js

```diff
@@ -55,6 +58,27 @@ var Mixin = {
   },
 
   /**
+   * @return {object} The queue to collect React async events.
+   */
+  getUpdateQueue: function() {
+    // We use a ReactNoopUpdateQueue that switches to ReactUpdateQueue while
+    // mounting
+    return Object.keys(ReactUpdateQueue).map(function(funcName) {
+      var funcObj = {};
+      funcObj[funcName] = function() {
+        return this._isInTransaction ?
+          ReactUpdateQueue[funcName].apply(ReactUpdateQueue, arguments) :
+          ReactNoopUpdateQueue[funcName]
+          .apply(ReactNoopUpdateQueue, arguments);
+      }.bind(this);
+      return funcObj;
+    }, this)
+    .reduce(function(ReactUpdate, funcObj) {
+      return Object.assign(ReactUpdate, funcObj);
+    });
```

Comment:
Not sure about this though, maybe there is a way to do better there. I personally would have used Immutable for this kind of trick, but, I understand the constraints here. So if I can do that differently, just tell me!